### PR TITLE
[2024 GPG KEY ROTATION] Rotate GPG key used to build the agent

### PR DESCRIPTION
**Pull request by Pythyu**
### What does this PR do?

Rotate the keys used to build the agent.
RPM keys are rotating from `fd4bf915` to `b01082d3`
APT keys are rotating from `ad9589b7` to `c0962c7d`
see the [Agent Signing Key Document](https://datadoghq.atlassian.net/wiki/spaces/BARX/pages/2971370900/Agent+signing+keys+history) to verify that we're replacing current keys by future keys

### Additional Notes

Renamed `DEB_GPG_KEY_NAME` because we aren't using subkeys with the new APT key.

### Describe how to test/QA your changes

For every RPM packages
  - `rpm --import https://keys.datadoghq.com/DATADOG_RPM_KEY_B01082D3.public`
  - `rpm --checksig <package>` -> We expect `digests signatures OK`

For every DEB packages
  - TBD

### CODE REVIEW

File: releasenotes/notes/2024-gpg-key-rotation-0aceffce7a9771fa.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+security:
+  - |
+    Current GPG keys used to sign new release of the agent are about to expire.
+    Following our 2024 GPG Key rotation plan, we rotated RPM and APT GPG keys used to sign the agent package.
```
Comment (drichards-87) [inline releasenotes/notes/2024-gpg-key-rotation-0aceffce7a9771fa.yaml:12]:
```suggestion
    Current GPG keys that are used to sign new releases of the Agent package are about to expire.
    Following our 2024 GPG key rotation plan, we rotated RPM and APT GPG keys.
```

**Review by drichards-87 — APPROVED** [review]
Left a suggestion from Docs and approved the PR.

**Review by amenasria — APPROVED** [review]
LGTM !

**Pull request merged**